feature/experimental/gribi/ate_tests/supervisor_failure/Supervisor_Failure_test.go

```diff
@@ -0,0 +1,347 @@
+
+}
+
+// configureATE configures port1 and port2 on the ATE.
+func configureATE(t *testing.T, ate *ondatra.ATEDevice) *ondatra.ATETopology {
+	top := ate.Topology().New()
+
+	p1 := ate.Port(t, "port1")
+	i1 := top.AddInterface(atePort1.Name).WithPort(p1)
+	i1.IPv4().
+		WithAddress(atePort1.IPv4CIDR()).
+		WithDefaultGateway(dutPort1.IPv4)
+
+	p2 := ate.Port(t, "port2")
+	i2 := top.AddInterface(atePort2.Name).WithPort(p2)
+	i2.IPv4().
+		WithAddress(atePort2.IPv4CIDR()).
+		WithDefaultGateway(dutPort2.IPv4)
+
+	return top
+}
+
+// testTraffic generates traffic flow from source network to
+// destination network via srcEndPoint to dstEndPoint and checks for
+// packet loss.
+func testTraffic(t *testing.T, ate *ondatra.ATEDevice, top *ondatra.ATETopology, srcEndPoint, dstEndPoint *ondatra.Interface) {
+	ethHeader := ondatra.NewEthernetHeader()
+	ipv4Header := ondatra.NewIPv4Header()
+	ipv4Header.DstAddressRange().
+		WithMin("203.0.113.0").
+		WithMax("203.0.113.254").
+		WithCount(250)
+
+	flow := ate.Traffic().NewFlow("Flow").
+		WithSrcEndpoints(srcEndPoint).
+		WithDstEndpoints(dstEndPoint).
+		WithHeaders(ethHeader, ipv4Header)
+
+	ate.Traffic().Start(t, flow)
+	time.Sleep(15 * time.Second)
+	ate.Traffic().Stop(t)
+
+	flowPath := ate.Telemetry().Flow(flow.Name())
+	if got := flowPath.LossPct().Get(t); got > 0 {
+		t.Errorf("LossPct for flow %s got %g, want 0", flow.Name(), got)
+	} else {
+		t.Logf("Traffic flows fine from ATE-port1 to ATE-port2")
+	}
+}
+
+// testArgs holds the objects needed by a test case.
+type testArgs struct {
+	ctx     context.Context
+	clientA *gribi.Client
+	dut     *ondatra.DUTDevice
+	ate     *ondatra.ATEDevice
+	top     *ondatra.ATETopology
+}
+
+// routeInstall configures a IPv4 entry through clientA. Ensure that the entry via ClientA
+// is active through AFT Telemetry.
+func routeInstall(ctx context.Context, t *testing.T, args *testArgs) {
+	// Add an IPv4Entry for 203.0.113.0/24 pointing to ATE port-2 via gRIBI-A,
+	// ensure that the entry is active through AFT telemetry
+	t.Logf("Add an IPv4Entry for %s pointing to ATE port-2 via gRIBI-A", ateDstNetCIDR)
+	args.clientA.AddNH(t, nhIndex, atePort2.IPv4, instance, fluent.InstalledInRIB)
+	args.clientA.AddNHG(t, nhgIndex, map[uint64]uint64{nhIndex: 1}, instance, fluent.InstalledInRIB)
+	args.clientA.AddIPv4(t, ateDstNetCIDR, nhgIndex, instance, "", fluent.InstalledInRIB)
+
+	// Verify the entry for 203.0.113.0/24 is active through AFT Telemetry.
+	ipv4Path := args.dut.Telemetry().NetworkInstance(instance).Afts().Ipv4Entry(ateDstNetCIDR)
+	if got, want := ipv4Path.Prefix().Get(t), ateDstNetCIDR; got != want {
+		t.Errorf("ipv4-entry/state/prefix got %s, want %s", got, want)
+	}
+	// Verify that static route(203.0.113.0/24) to ATE port-2 is preferred by the traffic.`
+	srcEndPoint := args.top.Interfaces()[atePort1.Name]
+	dstEndPoint := args.top.Interfaces()[atePort2.Name]
+	testTraffic(t, args.ate, args.top, srcEndPoint, dstEndPoint)
```

Comment:
This is redundant after verifying the entries using AFT Telemetry and FIB_ACK.

Reply:
per continuation from previous comment,  we ensure gRIBI programmed route and traffic flows,.

Reply:
Implemented.. New code push failed due to "Needs /gcbrun from a collaborator"..